Collector/DebugPluginDecorator.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+namespace Http\HttplugBundle\Collector;
+
+use Closure;
+use Http\Client\Common\Plugin;
+use Http\Client\Exception;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class DebugPluginDecorator implements Plugin
+{
+    /**
+     * @var Plugin
+     */
+    private $plugin;
+
+    /**
+     * @var Collector
+     */
+    private $collector;
+
+    /**
+     * @param Plugin $plugin
+     * @param Collector $collector
+     */
+    public function __construct(Plugin $plugin, Collector $collector)
+    {
+        $this->plugin = $plugin;
+        $this->collector = $collector;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function handleRequest(RequestInterface $request, callable $next, callable $first)
+    {
+        $profile = new Profile();
+        $profile->setPlugin($this->plugin);
+        $profile->setRequest($request);
+
+        $this->collector->getCurrentStack()->addProfile($profile);
+
+        $this->plugin->handleRequest($request, $this->wrap($next, $profile), $this->wrap($first, $profile));
+    }
+
+    /**
+     * @param callable $callable
+     * @param Profile  $profile
+     *
+     * @return Closure
+     */
+    private function wrap(callable $callable, Profile $profile)
+    {
+        return function (RequestInterface $request) use ($callable, $profile) {
+            return $callable($request)->then(function (ResponseInterface $response) use ($profile) {
```

Comment:
I get an error at this line. The `$callable($request)` call didn't return a Promise as I expected but null. I probably misunderstood something :/ As I said before, each plugin is decorated with a `DebugPluginDecorator` instance so I expected a Promise as specified by `Plugin::handleRequest`. Help is welcome here!